{
	"documenttitle": {
		"text": "Red Hat Security Advisory: python27:2.7 security, bug fix, and enhancement update",
		"lang": "en"
	},
	"DocumentType": "Security Advisory",
	"documentpublisher": {
		"type": "Vendor",
		"ContactDetails": "secalert@redhat.com",
		"IssuingAuthority": "Red Hat Product Security"
	},
	"documenttracking": {
		"identification": {
			"ID": "RHSA-2020:1605"
		},
		"Status": "Final",
		"Version": "1",
		"revisionhistory": {
			"revision": {
				"Number": "1",
				"Date": "2021-02-01T08:31:00Z",
				"Description": "Current version"
			}
		},
		"InitialReleaseDate": "2020-04-28T16:06:00Z",
		"CurrentReleaseDate": "2021-02-01T08:31:00Z",
		"generator": {
			"Engine": "Red Hat SDEngine 3.0.0",
			"Date": "2021-08-23T04:51:00Z"
		}
	},
	"documentnotes": {
		"note": [
			{
				"text": "\nAn update for the python27:2.7 module is now available for Red Hat Enterprise Linux 8.\n\nRed Hat Product Security has rated this update as having a security impact of Moderate. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE link(s) in the References section.\n    ",
				"title": "Topic",
				"type": "Summary",
				"ordinal": "1",
				"lang": "en"
			},
			{
				"text": "\nPython is an interpreted, interactive, object-oriented programming language that supports modules, classes, exceptions, high-level dynamic data types, and dynamic typing. The python27 packages provide a stable release of Python 2.7 with a number of additional utilities and database connectors for MySQL and PostgreSQL.\n\nThe following packages have been upgraded to a later upstream version: python2 (2.7.17). (BZ#1759944)\n\nSecurity Fix(es):\n\n* python-urllib3: Cross-host redirect does not remove Authorization header allow for credential exposure (CVE-2018-20060)\n\n* python: Cookie domain check returns incorrect results (CVE-2018-20852)\n\n* python-urllib3: CRLF injection due to not encoding the '\\r\\n' sequence leading to possible attack on internal service (CVE-2019-11236)\n\n* python-urllib3: Certification mishandle when error should be thrown (CVE-2019-11324)\n\n* python: email.utils.parseaddr wrongly parses email addresses (CVE-2019-16056)\n\n* python-requests: Redirect from HTTPS to HTTP does not remove Authorization header (CVE-2018-18074)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n\nAdditional Changes:\n\nFor detailed information on changes in this release, see the Red Hat Enterprise Linux 8.2 Release Notes linked from the References section.\n    ",
				"title": "Details",
				"type": "General",
				"ordinal": "2",
				"lang": "en"
			},
			{
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use",
				"type": "Legal Disclaimer",
				"ordinal": "3",
				"lang": "en"
			}
		]
	},
	"documentdistribution": {
		"text": "Copyright Â© 2021 Red Hat, Inc. All rights reserved.",
		"lang": "en"
	},
	"aggregateseverity": {
		"text": "Moderate",
		"namespace": "https://access.redhat.com/security/updates/classification/"
	},
	"documentreferences": {
		"reference": [
			{
				"type": "Self",
				"URL": "https://access.redhat.com/errata/RHSA-2020:1605",
				"Description": "https://access.redhat.com/errata/RHSA-2020:1605"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/security/updates/classification/#moderate",
				"Description": "https://access.redhat.com/security/updates/classification/#moderate"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.2_release_notes/index",
				"Description": "https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.2_release_notes/index"
			}
		]
	},
	"vulnerability": [
		{
			"ordinal": "1",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python-requests: Redirect from HTTPS to HTTP does not remove Authorization header",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-06-29T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2018-18074",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2020:1605"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2018-18074",
						"Description": "CVE-2018-18074"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1643829",
						"Description": "bz#1643829: CVE-2018-18074 python-requests: Redirect from HTTPS to HTTP does not remove Authorization header"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2018-10-10T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.2.0.GA:python27:2.7-8020020200117110429.90f98d4f"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "2",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python-urllib3: Cross-host redirect does not remove Authorization header allow for credential exposure",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-03-26T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2018-20060",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2020:1605"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2018-20060",
						"Description": "CVE-2018-20060"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1649153",
						"Description": "bz#1649153: CVE-2018-20060 python-urllib3: Cross-host redirect does not remove Authorization header allow for credential exposure"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2018-11-08T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.2.0.GA:python27:2.7-8020020200117110429.90f98d4f"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "3",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: Cookie domain check returns incorrect results",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-10-31T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2018-20852",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2020:1605"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2018-20852",
						"Description": "CVE-2018-20852"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1740347",
						"Description": "bz#1740347: CVE-2018-20852 python: Cookie domain check returns incorrect results"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-07-14T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.2.0.GA:python27:2.7-8020020200117110429.90f98d4f"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "4",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python-urllib3: CRLF injection due to not encoding the '\\r\\n' sequence leading to possible attack on internal service",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2019-03-13T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2019-11236",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2020:1605"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2019-11236",
						"Description": "CVE-2019-11236"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1700824",
						"Description": "bz#1700824: CVE-2019-11236 python-urllib3: CRLF injection due to not encoding the '\\r\\n' sequence leading to possible attack on internal service"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-04-15T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.2.0.GA:python27:2.7-8020020200117110429.90f98d4f"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "5",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python-urllib3: Certification mishandle when error should be thrown",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2019-04-17T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2019-11324",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2020:1605"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://www.openwall.com/lists/oss-security/2019/04/17/3",
						"Description": "https://www.openwall.com/lists/oss-security/2019/04/17/3"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2019-11324",
						"Description": "CVE-2019-11324"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1702473",
						"Description": "bz#1702473: CVE-2019-11324 python-urllib3: Certification mishandle when error should be thrown"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-04-18T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.2.0.GA:python27:2.7-8020020200117110429.90f98d4f"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "6",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: email.utils.parseaddr wrongly parses email addresses",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-07-19T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2019-16056",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2020:1605"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2019-16056",
						"Description": "CVE-2019-16056"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1749839",
						"Description": "bz#1749839: CVE-2019-16056 python: email.utils.parseaddr wrongly parses email addresses"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-09-06T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.2.0.GA:python27:2.7-8020020200117110429.90f98d4f"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "7",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: XSS vulnerability in the documentation XML-RPC server in server_title field",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2019-09-21T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2019-16935",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2020:1605"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2019-16935",
						"Description": "CVE-2019-16935"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1763229",
						"Description": "bz#1763229: CVE-2019-16935 python: XSS vulnerability in the documentation XML-RPC server in server_title field"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-09-27T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.2.0.GA:python27:2.7-8020020200117110429.90f98d4f"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		}
	],
	"producttree": {
		"xmlns": "http://www.icasi.org/CVRF/schema/prod/1.1",
		"branch": [
			{
				"type": "Product Family",
				"name": "Red Hat Enterprise Linux",
				"branch": [
					{
						"type": "Product Name",
						"name": "Red Hat Enterprise Linux AppStream (v. 8)",
						"fullproductname": {
							"text": "Red Hat Enterprise Linux AppStream (v. 8)",
							"productid": "AppStream-8.2.0.GA",
							"cpe": "cpe:/a:redhat:enterprise_linux:8::appstream"
						}
					}
				],
				"fullproductname": {}
			},
			{
				"type": "Product Version",
				"name": "python27:2.7-8020020200117110429.90f98d4f",
				"fullproductname": {
					"text": "python27-2.7-8020020200117110429.90f98d4f",
					"productid": "python27:2.7-8020020200117110429.90f98d4f",
					"cpe": "cpe:/a:redhat:enterprise_linux:8::appstream"
				}
			}
		],
		"relationship": [
			{
				"productreference": "python27:2.7-8020020200117110429.90f98d4f",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "AppStream-8.2.0.GA",
				"fullproductname": {
					"text": "python27:2.7-8020020200117110429.90f98d4f as a component of Red Hat Enterprise Linux AppStream (v. 8)",
					"productid": "AppStream-8.2.0.GA:python27:2.7-8020020200117110429.90f98d4f",
					"cpe": "cpe:/a:redhat:enterprise_linux:8::appstream"
				}
			}
		]
	}
}
